### PR TITLE
Update phpunit and allow php8

### PR DESCRIPTION
**Pull request by ChristophWurst**
This builds on https://github.com/php-opencloud/openstack/pull/324.

This also resolves all wrong method calls with the new phpunit and fixes and code/markup that triggers phpunit warnings otherwise. This will ease the upgrade to phpunit9.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
@jamiehannaford @haphan please let me know if this is ok the way I did it or if you want this to be split into smaller change sets. After my most recent commit CI should finally test the code on php8, so we'll see if anything needs to be adjusted in the code. The changes so far are merely to satisfy composer platform dependencies and updating test code for a new phpunit.

**Comment by ChristophWurst** [comment]
All green. Whohoo.

**Comment by haphan** [comment]
Let me know if this pr is ready for review. Thanks :)

**Comment by ChristophWurst** [comment]
IMO it is: the composer now allows php8, so consumers of the package can depend on it, CI and dependencies were updated accordingly :)

**Comment by ChristophWurst** [comment]
@haphan friends ping :grinning: 

if you want me to change or split anything in smaller junks just let me know :)

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +26,10 @@ before_install:
 before_script:
   - composer install --prefer-source
   - vendor/bin/parallel-lint --exclude vendor .
-  - vendor/bin/php-cs-fixer fix --dry-run --diff
+
+  # Once php8 is officially supported by the cs-fixer, the env variable should be dropped
+  # Ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702
+  - PHP_CS_FIXER_IGNORE_ENV=1 vendor/bin/php-cs-fixer fix --dry-run --diff
```
Comment (MorrisJobke) [inline .travis.yml:32]: php-cs-fixer 2.18 now supports PHP 8: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702#issuecomment-761957864

**Comment by ChristophWurst** [comment]
@jamiehannaford @haphan please review :)

**Comment by Elycin** [comment]
Hey there, I'm checking in for a status update on this library. 

This PR and #323 seem inactive for the past two months, so I figured that I could probably ask for an update. If there's any blockers in the way, is there something that I can do to help this project by any chance? There's a handful of things that depend on this and I'd like to contribute if possible. 

Best Regards,
Ely

**Comment by pcieszkowski** [comment]
Tried this PR on my project. Works really smooth.

**Comment by timmipetit** [comment]
Also tried this PR, works for us. Is there anything that is keeping it blocked from being merged?

**Review by jamiehannaford — APPROVED** [review]
Thanks for the contribution. I'm not actively maintaining this codebase anymore, but based on a quick review and feedback from other community members -- it seems this is fine to merge 👍

**Comment by jamiehannaford** [comment]
If anybody is interested in help maintain this project, please let me know. Otherwise we'll need to consider activating read-only mode.

**Comment by Elycin** [comment]
@jamiehannaford It may be worth adding a message to display in the package.json that will tell users installing the library that the package is looking for a new maintainer. I wish that I had the time to further contribute to the library myself, but I do hope that this library is maintained for how much I utilize it.

**Comment by jamiehannaford** [comment]
@Elycin Is this possible to do with Composer?  It's been a while since I've worked with PHP 😄

**Comment by Elycin** [comment]
@jamiehannaford I believe so. Found this on StackOverflow. https://stackoverflow.com/a/61254933

P.S.: can you also tag the release if possible?

**Comment by haphan** [comment]
Tagged v3.2.0

**Pull request merged**